snapcraft/internal/lifecycle.py

```diff
@@ -112,6 +118,42 @@ def execute(step, project_options, part_names=None):
             'type': config.data.get('type', '')}
 
 
+def _setup_core(deb_arch):
+    core_path = os.path.join(os.path.sep, 'snap', 'core', 'current')
+    if os.path.exists(core_path) and os.listdir(core_path):
+        logger.debug('{!r} already exists, skipping core setup'.format(
+            core_path))
+        return
+    snap_cache = SnapCache(project_name='snapcraft-core')
+
+    # Try to get the latest revision.
+    core_snap = snap_cache.get(deb_arch=deb_arch)
+    if core_snap:
+        # The current hash matches the filename
+        current_hash = os.path.splitext(os.path.basename(core_snap))[0]
+    else:
+        current_hash = ''
+
+    with tempfile.TemporaryDirectory() as d:
+        download_path = os.path.join(d, 'core.snap')
+        download_hash = snapcraft.download('core', 'stable', download_path,
+                                           deb_arch, but_hash=current_hash)
+        print('download', download_hash, 'current', current_hash)
+        if download_hash != current_hash:
+            snap_cache.cache(snap_filename=download_path)
+            snap_cache.prune(deb_arch=deb_arch, keep_hash=download_hash)
+
+    core_snap = snap_cache.get(deb_arch=deb_arch)
+
+    # Now unpack
+    logger.info('Setting up {!r} in {!r}'.format(core_snap, core_path))
```

Comment:
I will let you log a bug so we do the same thing for every place we already call sudo ;-)